exporter/cat1_view_helper.go

```diff
@@ -140,11 +177,29 @@ func valueOrDefault(val interface{}, def interface{}) interface{} {
 	return def
 }
 
-// conditional assignment. returns the second value only if the first value is zero
-// TODO: make arguments and return type interface{}. add "value is empty, zero, or nil" to description
-func condAssign(first int64, second int64) int64 {
-	if first != 0 {
-		return first
+// conditional assignment. returns the second value only if the first value is empty, zero, or nil
+func condAssign(first interface{}, second interface{}) interface{} {
+	switch val := first.(type) {
+	case string:
+		if val != "" {
+			return first
+		}
+	case int64:
+		if val != 0 {
+			return first
+		}
+	case int:
+		if val != 0 {
+			return first
+		}
+	case *int64:
+		if val != nil {
+			return first
+		}
+	case *int:
+		if val != nil {
+			return first
+		}
 	}
 	return second
```

Comment:
Sorry if I wasn't clear but with my last comment on this line, I meant I think it would be better style to have `return second` in the `default` switch case and not outside of the switch.